api/src/services/__tests__/programsService.ts

```diff
@@ -0,0 +1,372 @@
+      const curriculumActivityId = 4;
+
+      const [, , , matchingCurriculumActivity] = curriculumActivitiesList;
+
+      mockQuery(
+        'select `id`, `title`, `description_text`, `curriculum_week`, `curriculum_day`, `start_time`, `end_time`, `duration`, `activity_type_id`, `curriculum_id` from `activities` where `id` = ?',
+        [curriculumActivityId], // binding param // replace above ? with the value // findProgram() get the first program object, so the limit is 1
+        [matchingCurriculumActivity]
+      );
+
+      expect(await findCurriculumActivity(curriculumActivityId)).toEqual(
+        matchingCurriculumActivity
+      );
+    });
+  });
+
+  describe('listProgramActivities', () => {
+    it('should list all available activities for the specified program', async () => {
+      const programId = 2;
+      const curriculumId = 2;
+
+      const [, matchingProgram] = programsList;
+
+      const matchingCurriculumActivities: CurriculumActivity[] = [
+        curriculumActivitiesList[3],
+        curriculumActivitiesList[4],
+        curriculumActivitiesList[5],
+      ];
+
+      const matchingProgramActivities: ProgramActivity[] = [
+        program2ActivitiesList[0],
+        program2ActivitiesList[1],
+        program2ActivitiesList[2],
+      ];
+
+      // findProgram
+      mockQuery(
+        'select `id`, `title`, `start_date`, `end_date`, `time_zone`, `curriculum_id` from `programs` where `id` = ?',
+        [programId],
+        [matchingProgram]
+      );
+
+      // listCurriculumActivities
+      mockQuery(
+        'select `id`, `title`, `description_text`, `curriculum_week`, `curriculum_day`, `start_time`, `end_time`, `duration`, `activity_type_id`, `curriculum_id` from `activities` where `curriculum_id` = ?',
+        [curriculumId],
+        matchingCurriculumActivities
+      );
+
+      // activity types
+      mockQuery(
+        'select `id`, `title` from `activity_types`',
+        [],
+        activityTypesList
+      );
+
+      // listProgramActivities
+      expect(await listProgramActivities(programId)).toEqual(
+        matchingProgramActivities
+      );
+    });
+  });
+
+  describe('findProgramWithActivities', () => {
+    it('should give the details of the specified program including its activities', async () => {
+      const programId = 2;
+
+      const [, matchingProgram] = programsList;
+
+      const matchingCurriculumActivities: CurriculumActivity[] = [
+        curriculumActivitiesList[3],
+        curriculumActivitiesList[4],
+        curriculumActivitiesList[5],
+      ];
+
+      const matchingProgramActivities: ProgramActivity[] = [
+        program2ActivitiesList[0],
+        program2ActivitiesList[1],
+        program2ActivitiesList[2],
+      ];
+
+      const programWithActivities = JSON.parse(JSON.stringify(matchingProgram));
+      programWithActivities.activities = matchingProgramActivities;
+
+      // findProgram
```

Comment:
Comments can also be removed from lines 346 - 361 for same reason above.

Reply:
Extraneous comments have been removed.